rs/types/types/src/messages/inter_canister.rs

```diff
@@ -653,6 +655,57 @@ impl From<Response> for RequestOrResponse {
     }
 }
 
+/// A message that acts as a blocker for the message behind it in the `StreamIndexedQueue` of a
+/// `StreamSlice` until the `begin` in a different `StreamSlice` coming from `subnet_id` is verified
+/// to be >= `index`.
+#[derive(Clone, Debug, Eq, PartialEq, Hash, Deserialize, Serialize)]
+#[cfg_attr(test, derive(ExhaustiveSet))]
+pub struct StreamBlocker {
+    pub subnet_id: SubnetId,
+    pub index: StreamIndex,
+}
+
+/// `RequestOrResponse` extended by `StreamBlocker` which exists only in stream slices and streams.
+#[derive(Clone, Debug, Eq, PartialEq, Hash, Deserialize, Serialize)]
+#[cfg_attr(test, derive(ExhaustiveSet))]
+pub enum RequestOrResponseOrBlocker {
+    Request(Arc<Request>),
+    Response(Arc<Response>),
+    Blocker(Arc<StreamBlocker>),
+}
+
+impl From<&RequestOrResponse> for RequestOrResponseOrBlocker {
+    fn from(msg: &RequestOrResponse) -> RequestOrResponseOrBlocker {
+        match msg {
+            RequestOrResponse::Request(request) => {
+                RequestOrResponseOrBlocker::Request(request.clone())
+            }
+            RequestOrResponse::Response(response) => {
+                RequestOrResponseOrBlocker::Response(response.clone())
+            }
+        }
+    }
+}
+
+/// Convenience `CountBytes` implementation that returns the same value as
+/// `RequestOrResponseOrBlocker::Blocker(self).count_bytes()`, so we don't need to wrap
+/// `self` into a `RequestOrResponseOrBlocker` only to calculate its estimated byte size.
+impl CountBytes for StreamBlocker {
+    fn count_bytes(&self) -> usize {
+        size_of::<RequestOrResponseOrBlocker>() + size_of::<StreamBlocker>()
+    }
+}
+
+impl CountBytes for RequestOrResponseOrBlocker {
+    fn count_bytes(&self) -> usize {
+        match self {
+            RequestOrResponseOrBlocker::Request(req) => req.count_bytes(),
+            RequestOrResponseOrBlocker::Response(resp) => resp.count_bytes(),
+            RequestOrResponseOrBlocker::Blocker(blo) => blo.count_bytes(),
```

Comment:
Nit: `req` and `resp` are common abbreviations for `request` and `response`. This is not the case for `blo`. Nor is it `blocker` long enough to require abbreviating.